src/librustc_mir/interpret/memory.rs

```diff
@@ -791,6 +791,25 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         self.get_raw(ptr.alloc_id)?.read_c_str(self, ptr)
     }
 
+    /// Reads a 0x0000-terminated sequence of bytes from memory. Returns them as a slice.
+    /// Needed for reading wide-strings in Windows-OS
+    pub fn read_wide_str(&self, ptr: Scalar<M::PointerTag>) -> InterpResult<'tcx, &[u8]> {
+        let widestr_u8_initbyte = self.read_bytes(ptr, Size::from_bytes(1))?;
+        let mut widestr_len = 0; // length in bytes
+        // The below unsafe block uses a raw-pointer(*const u8) to find the length
+        // of the wide-string, which terminates with a double null byte.
+        // The below unsafe block doesn't write anything to Memory.
+        // Each character in a wide-string takes up two bytes.
+        unsafe {
+            let mut tracker = &widestr_u8_initbyte[0] as *const u8;
+            while !(*tracker == 0 && *tracker.add(1) == 0) {
+                tracker = tracker.add(2);
```

Comment:
This is actually incorrect `unsafe` code. You are taking a slice of length 1, turning it into a raw pointer, and reading *outside the bounds of that slice*. That's UB due to violating Rust's aliasing rules.

Reply:
Even worse, isn't this going out-of-bounds and causing dangling pointer deref's in case there is no trailing double-NULL?